### PR TITLE
feat: allow union and ref field updates for reverse fields

### PR DESCRIPTION
**Pull request by joelpierre**
PR allows us to update reverse fields on union and ref fields.

@notrab - Let me know if this looks like it touches the right areas?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,7 +53,7 @@
     "size-limit": "4.10.1",
     "tsdx": "0.14.1",
     "tslib": "2.1.0",
-    "typescript": "4.2.3"
+    "typescript": "4.5.5"
```
Comment (joelpierre) [inline package.json:56]: Cheeky version bump to the latest.... didn't see any harm 😬

File: src/model.ts [hunk 2]
```diff
@@ -302,21 +303,25 @@ class ModelClass implements Model, ChangeItem {
     }
 
     const { type, ...fieldChanges } = fieldArgs;
-    const field = new Field(fieldChanges, MutationMode.Update);
+    const field = new Field(fieldChanges as FieldArgs, MutationMode.Update);
```
Comment (joelpierre) [inline src/model.ts:286]: This was giving a TS error 🤔 . Even though `fieldArgs` is "any".

File: src/model.ts [hunk 3]
```diff
@@ -302,21 +303,25 @@ class ModelClass implements Model, ChangeItem {
...
     this.listener.registerChange(field);
     return this;
   }
 
   addRelationalField(passedFieldArgs: any): Model {
     const fieldArgs = { ...passedFieldArgs };
     fieldArgs.modelApiId = this.args.apiId;
+
+    const fieldTypeUpper = fieldArgs.type?.toUpperCase();
+    const fieldModelUpper = fieldArgs.model?.toUpperCase();
```
Comment (joelpierre) [inline src/model.ts:296]: The strings were making me anxious considering `passedFieldArgs` is of type any 😅 . Also its nice to normalise string match checks.

File: src/model.ts (hunk 2 above)
Comment (joelpierre) [reply src/model.ts:286]: Screenshot 2022-02-23 at 07 35 59

**Review by flexzuu — APPROVED** [review]

**Pull request merged**